### PR TITLE
#1878 - UI for the boolean field

### PR DESCRIPTION
**Pull request by EthanW96**
As I mention in the issue we have some additional features that might be helpful.

- Ability to enter specific words into the "Yes" and "No" fields. Sometimes different words convey meaning better for boolean fields. I'm also wondering if a rework like this could help with language translation options?
- Ability to set a default instead of defaulted to "No".
- Ability to pick a checkbox only (no language) or Yes/No dropdown

Here is the result of the current commit:

image

### CODE REVIEW

**Comment by EthanW96** [comment]
@corsacca nice that's awesome! Should I set this Boolean field up as a component somewhere? Or is this issue on hold till the refactor is complete?

File: dt-core/admin/menu/tabs/tab-custom-fields.php [hunk 1]
```diff
@@ -1538,6 +1540,15 @@ private function process_add_field( $post_submission ){
                     'customizable' => 'all',
                     'private' => $field_private,
                 ];
+            } elseif ( $field_type === 'boolean' ){
+                $new_field = [
+                    'name' => $post_submission['new_field_name'],
+                    'default' => [],
```
Comment (corsacca) [inline dt-core/admin/menu/tabs/tab-custom-fields.php:1546]: @EthanW96 default here should probably be `false`

Comment (EthanW96) [reply dt-core/admin/menu/tabs/tab-custom-fields.php:1546]: Cool, I've adjusted this be marked as false by default.

**Comment by EthanW96** [comment]
All sounds good to me. I did wonder how the display of the boolean field worked so seamlessly without additional code written on my part. I'm still learning where everything is located in the codebase. It's good to know that https://github.com/DiscipleTools/disciple-tools-theme/commit/8e1ab90bc14a43e62d7f8400c1e13c24eada90d0 exists.

**Comment by EthanW96** [comment]
@corsacca I realized while looking at this line:
https://github.com/DiscipleTools/disciple-tools-theme/blob/8e1ab90bc14a43e62d7f8400c1e13c24eada90d0/dt-core/global-functions.php#L599
The display of the boolean field doesn't actually handle a default (I could enter anything in the review you pointed out above) so https://github.com/DiscipleTools/disciple-tools-theme/pull/2288/commits/54b2064eacf536e9556da2d69117aa34d6ac255b adds that functionality. When the default is being set it must be a strict `true` or `false`.

**Comment by corsacca** [comment]
Implementing the default might be a bit tricky.

I added this field in contacts/base-setup.php

```
    $fields['bool_test'] = [
        'name' => __( 'Boolean Test', 'disciple_tools' ),
        'type' => 'boolean',
        'default' => true,
        'tile' => 'status',
    ];
```
with default is true.

It shows the field with YES selected:
image

But, when we change the value to NO and refresh the page it show YES.
This is because the `!empty()`  is figuring out the selected with show false, so then the "default" is chosen.

**Comment by EthanW96** [comment]
Good call out. I've made some adjustments to that logic and to the modular list so boolean fields that are defaulted to true show correctly. Are there other places in D.T. that boolean fields may appear that need additional logic for defaults?

**Pull request merged**